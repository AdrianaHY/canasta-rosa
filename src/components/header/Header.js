import React, { Component } from 'react';

import LogoDesktop from '../../assets/logo_desktop.svg';
import LogoMobile from '../../assets/logo_mobile.svg';
import HamburgerMenu from '../../assets/hamburguer-menu.svg';

import './Header.scss';

class Header extends Component {
	render() {
		const { handleClick } = this.props;
		return(
			<div className="header-container">
				<nav className="header">
					<div className="logo-container">
						<img className="logo-desktop" src={LogoDesktop} alt="logo-desktop" />
						<img className="logo-mobile" src={LogoMobile} alt="logo-mobile" />
						<button onClick={handleClick} className="menu">
							CATEGOR√çAS <span className="chevron bottom" />
						</button>
						<button onClick={handleClick} className="hamburger-menu">
							<img className="image-burger" src={HamburgerMenu} alt="hamburger-menu"/>
						</button>
					</div>
					<ul className="menu-items">
						<li className="menu-item">MARKET</li>
						<li className="menu-item">INSPIRE</li>
						<li className="menu-item">PROP</li>
					</ul>
				</nav>
			</div>
		)
	}
}

export default Header;

