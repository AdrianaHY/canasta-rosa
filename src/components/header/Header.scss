@import '../../styles/_variables.scss';

.header {
	display: flex;
	flex-direction: row;
	justify-content: space-between;
	align-items: center;
	font-family: $fontFamily;
	background: #f2f2f2;
	padding: 10px 0 10px 15px;

	.menu {
		display: none;

		@include breakpoint ('large') {
			display: flex;
			padding-left: $paddingLarge;
			border: none;
			background: transparent;
			font-size: 16px;
		}
	}

	.hamburger-menu {
		display: flex;
		padding-left: 70%;

		@include breakpoint ('large') {
			display: none;
		}
	}

	.image-burger {
		height: 30px;
	}

	.logo-container {
		display: flex;
	}

	.logo-mobile{
		display: flex;

		@include breakpoint ('large') {
			display: none;
		}
	}

	.logo-desktop {
		display: none;

		@include breakpoint ('large') {
			display: flex;
		}
	}

	.chevron::before {
		border-style: solid;
		border-width: 0.25em 0.25em 0 0;
		content: '';
		display: inline-block;
		height: 0.45em;
		left: 0.15em;
		position: relative;
		top: 0.15em;
		transform: rotate(-45deg);
		vertical-align: top;
		width: 0.45em;
	}

	.bottom:before {
		top: 0;
		transform: rotate(135deg);
		margin-left: $paddingTiny;
	}

	.right:before {
		left: 0;
		transform: rotate(45deg);
		border-width: 0.10em 0.10em 0 0;
	}

	.menu-items {
		display: none;
		list-style: none;
		padding-right: $paddingLarge * 2;

		@include breakpoint ('large') {
			display: flex;
		}
	}

	.menu-item {
		padding: 0 $paddingSmall;
	}
}

.container-categories {
	.categories {
		list-style: none;
		font-family: 'Sarabun', sans-serif;
		font-size: 18px;
		display: flex;
		flex-direction: column;
		width: 100%;
		padding: 0 $paddingMedium;

		@include breakpoint ('large') {
			padding: $paddingMedium $paddingLarge;
		}
	}

	.categorie {
		width: auto;
		display: flex;
		align-items: center;
		justify-content: space-between;
		color: #37373E;
		padding: ($paddingTiny+$paddingSmall) 0;

		@include breakpoint ('large') {
			width: 600px;
		}
	}

	.list-of-categories {
		display: none;
	}

	.show-list {
		display: flex;
		justify-content: space-between;
	}

	.chevron::before {
		border-style: solid;
		border-width: 0.25em 0.25em 0 0;
		content: '';
		display: inline-block;
		height: 0.45em;
		left: 0.15em;
		position: relative;
		top: 0.15em;
		transform: rotate(-45deg);
		vertical-align: top;
		width: 0.45em;
	}

	.right:before {
		left: 0;
		transform: rotate(45deg);
		border-width: 0.10em 0.10em 0 0;
	}
}

