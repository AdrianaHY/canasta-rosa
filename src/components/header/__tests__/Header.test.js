import React from 'react';
import { shallow } from 'enzyme';
import Header from '../Header';

describe('components | header | Header', () => {
	it('Renders the component', () => {
		const props = {
			handleClick: jest.fn()
		};

		const component= shallow(
			<Header { ...props } />
		);

		expect(component).toMatchSnapshot();
	})
});