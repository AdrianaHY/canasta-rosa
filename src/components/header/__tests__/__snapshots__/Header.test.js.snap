// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | header | Header Renders the component 1`] = `ShallowWrapper {}`;
